packages/angular/ssr/src/app-engine.ts

```diff
@@ -24,23 +23,23 @@ import { stripIndexHtmlFromURL, stripTrailingSlash } from './utils/url';
  */
 export class AngularAppEngine {
   /**
-   * Hooks for extending or modifying the behavior of the server application.
-   * These hooks are used by the Angular CLI when running the development server and
-   * provide extensibility points for the application lifecycle.
+   * A flag to enable or disable the rendering of prerendered routes.
+   *
+   * Typically used during development to avoid prerendering all routes ahead of time,
+   * allowing them to be rendered on the fly as requested.
    *
    * @private
    */
-  static ɵhooks = /* #__PURE__*/ new Hooks();
+  static ɵallowStaticRouteRender = false;
```

Comment:
Consider: Should this default to `true`? If someone naiively uses this class without any extra context, they would likely expect static rendering to work. It's only specific dev use cases where a user would want to go out of their way to disable static rendering.

Reply:
This is actually to allow static pages to be renderer, not served. 

Only in the dev-server and builder should be done.